File: php/lib/steam/community/l4d/L4DWeapon.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

require_once STEAM_CONDENSER_PATH . 'steam/community/l4d/AbstractL4DWeapon.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage Steam Community
 */
class L4DWeapon extends AbstractL4DWeapon {

    /**
     * Creates a new instance of L4DWeapon based on the assigned XML data
     * @param $weaponData
     */
    public function __construct($weaponData) {
        parent::__construct($weaponData);

        $this->killPercentage      = (string) $weaponData->killpct;
    }
}
?>

File: ruby/lib/steam/community/l4d/l4d_weapon.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2009-2011, Sebastian Staudt

require 'steam/community/l4d/abstract_l4d_weapon'

# This class represents the statistics of a single weapon for a user in
# Left4Dead
#
# @author Sebastian Staudt
class L4DWeapon

  include AbstractL4DWeapon

  # Creates a new instance of a weapon based on the given XML data
  #
  # @param [REXML::Element] weapon_data The XML data for this weapon
  def initialize(weapon_data)
    super weapon_data

    @kill_percentage = weapon_data.elements['killpct'].text.to_f * 0.01
  end

end
